mlir/lib/Target/LLVMIR/Dialect/LLVMIR/LLVMToLLVMIRTranslation.cpp

```diff
@@ -271,6 +271,31 @@ static void convertLinkerOptionsOp(ArrayAttr options,
   linkerMDNode->addOperand(listMDNode);
 }
 
+static llvm::Metadata *
+convertModuleFlagValue(StringRef key, ArrayAttr arrayAttr,
+                       llvm::IRBuilderBase &builder,
+                       LLVM::ModuleTranslation &moduleTranslation) {
+  llvm::LLVMContext &context = builder.getContext();
+  llvm::MDBuilder mdb(context);
+  SmallVector<llvm::Metadata *> nodes;
+
+  if (key == "CG Profile") {
```

Comment:
Nit: Can you change this to an early return. Also, do we have this string key available in some header?

Reply:
It's already returning right before the if scope ends, can you elaborate? If you mean returning early if the name isn't "CG Profile": my next PR will introduce yet another of these, that's why I didn't do it.

> do we have this string key available in some header?

Nope. Any ideas where to put this?

Reply:
Not sure if there is a great place. In the LLVMDialect.td we define a couple of attribute names. Maybe this could be extended with metadata keywords if there are many more. 

In theory we could also use an I32EnumAttr. Tablegen generates symbolize and stringify functions for them to convert between string and enum. However, the module flag attr then stores a string attribute. So I am unsure if that makes sense. Maybe if there are many more keywords?

LLVM seems to use the string literal everywhere so there is not really a way to generate an LLVM enum attr...

Reply:
Added:
```
    /// Names of known llvm module flag keys.
    static StringRef getModuleFlagKeyCGProfileName() {
      return "CG Profile";
    }
```
To `LLVMDialect.td`, we do not really have an attribute for it (only for the entries) so I refrained from using "Attr" anywhere in the name.

Reply:
Nice!